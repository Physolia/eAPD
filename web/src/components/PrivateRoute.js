import PropTypes from 'prop-types';
import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

const PrivateRoute = ({ authenticated, component: Component, ...rest }) => (
  <Route
    {...rest}
    render={props =>
      authenticated ? (
        <Component {...props} />
      ) : (
        <Redirect
          to={{
            pathname: '/login',
            state: { from: props.location }
          }}
        />
      )
    }
  />
);

PrivateRoute.propTypes = {
  authenticated: PropTypes.bool.isRequired,
  component: PropTypes.elementType.isRequired,
  location: PropTypes.object.isRequired
};

const mapStateToProps = ({ auth: { authenticated } }) => ({ authenticated });

export default connect(mapStateToProps)(PrivateRoute);

export { PrivateRoute as plain, mapStateToProps };
