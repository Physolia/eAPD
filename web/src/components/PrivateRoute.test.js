import { shallow } from 'enzyme';
import React from 'react';

import { plain as PrivateRoute, mapStateToProps } from './PrivateRoute';

describe('private route component', () => {
  const props = {
    authenticated: false,
    location: { from: 'over there' },
    component: () => {}
  };

  test('renders correctly if logged out', () => {
    const component = shallow(<PrivateRoute {...props} />);
    expect(component).toMatchSnapshot();
    expect(component.prop('render')(props)).toMatchSnapshot();
  });

  test('renders correctly if logged in', () => {
    const component = shallow(<PrivateRoute {...props} authenticated />);
    expect(component).toMatchSnapshot();
    expect(
      component.prop('render')({ ...props, authenticated: true })
    ).toMatchSnapshot();
  });

  test('maps state to props', () => {
    const state = {
      auth: {
        authenticated: 'some value'
      }
    };

    expect(mapStateToProps(state)).toEqual({ authenticated: 'some value' });
  });
});
