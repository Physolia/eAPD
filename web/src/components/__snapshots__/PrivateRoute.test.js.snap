// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`private route component renders correctly if logged in 1`] = `
<Route
  location={
    Object {
      "from": "over there",
    }
  }
  render={[Function]}
/>
`;

exports[`private route component renders correctly if logged in 2`] = `
<component
  authenticated={true}
  component={[Function]}
  location={
    Object {
      "from": "over there",
    }
  }
/>
`;

exports[`private route component renders correctly if logged out 1`] = `
<Route
  location={
    Object {
      "from": "over there",
    }
  }
  render={[Function]}
/>
`;

exports[`private route component renders correctly if logged out 2`] = `
<Redirect
  to={
    Object {
      "pathname": "/login",
      "state": Object {
        "from": Object {
          "from": "over there",
        },
      },
    }
  }
/>
`;
